sklearn/metrics/scorer.py

```diff
@@ -44,7 +45,69 @@
 from ..base import is_regressor
 
 
-class _BaseScorer(metaclass=ABCMeta):
+def _method_caller(cache, estimator, method, *args, **kwargs):
```

Comment:
should this be called something like ``_cached_call``? It's a function so a verb seems better and I'd like it to contain ``cache`` because that's what it's for.